src/basic/path-util.c

```diff
@@ -853,3 +854,28 @@ char *prefix_root(const char *root, const char *path) {
         strcpy(p, path);
         return n;
 }
+
+bool path_prefix_in_list(const char *path, const char *x, ...) {
+        const char *p;
+        va_list ap;
+
+        if (!path || !x)
+                return false;
+
+        va_start(ap, x);
+
+        if (path_startswith(path, x)) {
+                va_end(ap);
+                return true;
+        }
+
+        while ((p = va_arg(ap, const char*)))
+                if (path_startswith(path, p)) {
+                        va_end(ap);
+                        return true;
+                }
+
+        va_end(ap);
+
+        return false;
+}
```

Comment:
This seems overly complicated, no? Why treat `x` special, and not just walk the list? If it's empty, the loop will be a noop. Also, you can have a local bool variable called `found` or something, and if `path_startswith()` returns success, just latch it and break from the loop. That would make the code contain just one `va_end()`. Makes sense, or do I miss anything?